$NetBSD: patch-aa,v 1.2 2011/12/22 04:33:28 dholland Exp $

- newer lex does not define YY_PROTO
- hack around issue with date parsing

--- src/mlex.l.orig	2004-08-13 11:18:23.000000000 +0000
+++ src/mlex.l
@@ -66,7 +66,7 @@
  ****************************************************************************/
   
 #define EXECUTE(fun) ((fun) ? fun (), 0 : 0)
-#define YY_DECL  int mlex_scan YY_PROTO ((void))
+#define YY_DECL  int mlex_scan(void)
 #define YY_BREAK offset += yyleng; break;
 
 
@@ -352,8 +352,14 @@ TEXT_PLAIN_CHARSET [ \t]*text\/plain.*(;
                  * obsolete one first, and check if it was possible to get
                  * the date
                  */
+		/** FIXME FIXME FIXME:
+		 * A more sophisticated Date: parser is needed, the first
+		 * call _never_ returns NULL (at least) on *BSD. Therefore
+		 * disable support for two digit years in order to NOT
+		 * garble all correctly formatted date headers....
+
                 tmp = strptime (date, "%d %b %y %H:%M:%S", &tm);
-                if (tmp == NULL)
+                if (tmp == NULL) */
                         tmp = strptime (date, "%d %b %Y %H:%M:%S", &tm);
 
 #ifdef HAVE_LOCALE_H
